### PR TITLE
feat: Move backoff to to the tracker to remove errors

### LINKED ISSUE #318: Backoff in Tracker per Allocation
Currently, backoff is being thrown as an error and it's becoming hard to understand what is the real error causing the backoff.

We want to create the backoff inside the tracker so that in case there's an error, that allocation is not selected until the backoff period is over.

### PR DESCRIPTION
**Pull request by carlosvdr**
fixes #318

### CODE REVIEW

File: tap-agent/src/agent/sender_account.rs [hunk 1]
```diff
@@ -1082,20 +1097,21 @@ pub mod tests {
...
-                    triggered_rav_request: triggered_rav_request.clone(),
+                    triggered_rav_request,
                     receipts: Arc::new(Mutex::new(Vec::new())),
                     next_rav_value: Arc::new(Mutex::new(0)),
-                    next_unaggregated_fees_value: unaggregated_fees.clone(),
+                    next_unaggregated_fees_value: unaggregated_fees,
                 },
-                triggered_rav_request,
-                unaggregated_fees,
+                Ok(mock_rav_request_response),
```
Comment (suchapalaver) [inline tap-agent/src/agent/sender_account.rs:1114]: Why are we returning a `Result` here? I couldn't an error we're trying to handle 🤔

File: tap-agent/src/agent/sender_fee_tracker.rs [hunk 2]
```diff
@@ -40,6 +40,21 @@ pub struct SenderFeeTracker {
...
     blocked_addresses: HashSet<Address>,
+    failed_ravs: HashMap<Address, FailedRavInfo>,
+}
+
+#[derive(Debug, Clone)]
+pub struct FailedRavInfo {
+    failed_ravs_count: u32,
+    failed_rav_backoff_time: Instant,
+}
+impl Default for FailedRavInfo {
```
Comment (suchapalaver) [inline tap-agent/src/agent/sender_fee_tracker.rs:51]:
```suggestion

impl Default for FailedRavInfo {
```

**Comment by suchapalaver** [comment]
@carlosvdr The main thing looks good to me but I'm going to let @gusinacio take a look 👍

File: tap-agent/src/agent/sender_account.rs (hunk 1 above)
Comment (carlosvdr) [reply tap-agent/src/agent/sender_account.rs:1114]: Did so to make it more like the [real response](https://github.com/graphprotocol/indexer-rs/blob/f6ebdd8e1b1441e927cc202d9510ff8edba652c7/tap-agent/src/agent/sender_account.rs#L220), where it also responds with a result

File: tap-agent/src/agent/sender_fee_tracker.rs [hunk 3]
```diff
@@ -288,4 +322,35 @@ mod tests {
         assert_eq!(tracker.get_total_fee_outside_buffer(), 0);
         assert_eq!(tracker.get_total_fee(), 0);
     }
+
+    #[test]
+    fn test_filtered_backed_off_allocations() {
+        let allocation_id_0 = address!("abababababababababababababababababababab");
+        let allocation_id_1 = address!("bcbcbcbcbcbcbcbcbcbcbcbcbcbcbcbcbcbcbcbc");
+        const BUFFER_WINDOW: Duration = Duration::from_secs(61);
```
Comment (gusinacio) [inline tap-agent/src/agent/sender_fee_tracker.rs:330]: Hey, I don't like that this is 61 seconds. This is sleeping, and it means that we need to wait 61 seconds for tests to pass.

Maybe you could update this to be really small, so it's not that problematic.

Also, could you update the name to BACKOFF_SLEEP_DURATION so it's clear what it does.

File: tap-agent/src/agent/sender_account.rs [hunk 4]
```diff
@@ -1196,8 +1212,9 @@ pub mod tests {
         ActorRef<SenderAllocationMessage>,
         JoinHandle<()>,
     ) {
-        let (mock_sender_allocation, triggered_rav_request, next_unaggregated_fees) =
+        let (mock_sender_allocation, rav_request_result) =
             MockSenderAllocation::new_with_triggered_rav_request();
+        let (triggered_rav_request, next_unaggregated_fees) = rav_request_result.unwrap();
```
Comment (gusinacio) [inline tap-agent/src/agent/sender_account.rs:1217]: You are unwrapping the result from `new_with_triggered_rav_request`. Isn't removing the result better?

**Review by gusinacio — CHANGES_REQUESTED** [review]

File: tap-agent/src/agent/sender_account.rs (hunk 1 above)
Comment (carlosvdr) [reply tap-agent/src/agent/sender_account.rs]: yeah removed it afterall

**Pull request merged**